### PR TITLE
feat(capture-sdk): Injected toolbar customisation

### PR DESCRIPTION
**Pull request by AdnanYupi**
PIA-2872

### CODE REVIEW

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/camera/CameraFragmentImpl.java [hunk 1]
```diff
@@ -752,6 +760,38 @@ private void bindViews(final View view) {
...
+    private void setTopBarInjectedViewContainer() {
+        if (GiniCapture.hasInstance()) {
+            topAdapterInjectedViewContainer.setInjectedViewAdapter(GiniCapture.getInstance().getNavigationBarTopAdapter());
+
+            if (topAdapterInjectedViewContainer.getInjectedViewAdapter() == null)
+                return;
+
+            if (mFragment.getActivity() == null)
+                return;
+
```
Comment (a-szotyori) [inline capture-sdk/sdk/src/main/java/net/gini/android/capture/camera/CameraFragmentImpl.java:776]: [Request] Back button click listener is missing.

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/camera/CameraFragmentImpl.java [hunk 2]
```diff
@@ -752,6 +760,38 @@ private void bindViews(final View view) {
...
+            topAdapterInjectedViewContainer.getInjectedViewAdapter().setNavButtonType(NavButtonType.BACK);
+            topAdapterInjectedViewContainer.getInjectedViewAdapter().setTitle(mFragment.getActivity().getResources().getString(R.string.gc_title_camera));
+            topAdapterInjectedViewContainer.getInjectedViewAdapter().setMenuResource(R.menu.gc_camera);
+            topAdapterInjectedViewContainer.getInjectedViewAdapter().setOnMenuItemClickListener(item -> {
+                startHelpActivity();
```
Comment (a-szotyori) [inline capture-sdk/sdk/src/main/java/net/gini/android/capture/camera/CameraFragmentImpl.java:781]: [Request] Please check the click item's id and only start help activity, if it's the right id. If it's an unknown id, then I propose to throw an `UnsupportedOperationException` with a message that says something like "Unknown menu item id. Please don't call our OnMenutItemClickListener for custom menu items.".

This way we prevent clients calling our `OnMenuItemClickListener` for their custom menu items. If they use custom menu items, then they should call our click listener only for the menu items we provided. Dealing with their custom menu items is not our problem. 🙂

**Review by a-szotyori — CHANGES_REQUESTED** [review]
Nice! I have only two small requests.

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/camera/CameraFragmentImpl.java (hunk 2 above)
Comment (AdnanYupi) [reply capture-sdk/sdk/src/main/java/net/gini/android/capture/camera/CameraFragmentImpl.java]: Done

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/camera/CameraFragmentImpl.java (hunk 1 above)
Comment (AdnanYupi) [reply capture-sdk/sdk/src/main/java/net/gini/android/capture/camera/CameraFragmentImpl.java:776]: Done

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/view/NavigationBarTopAdapter.kt [hunk 3]
```diff
@@ -69,6 +77,8 @@ internal class DefaultNavigationBarTopAdapter : NavigationBarTopAdapter {
 
     var viewBinding: GcNavigationBarTopBinding? = null
 
+    constructor()
+
```
[the remark this answers is not in the document]
Comment (AdnanYupi) [reply capture-sdk/sdk/src/main/java/net/gini/android/capture/view/NavigationBarTopAdapter.kt]: Unnecessary constructor removed it

**Pull request merged**